src/schemes/boundary/open_boundary/system.jl

```diff
@@ -548,23 +548,34 @@ end
 @inline function transform_particle!(fluid_system::FluidSystem, system,
                                      boundary_zone, particle, v, u, v_fluid, u_fluid)
     # Activate particle in boundary zone
-    activate_particle!(system, particle, v, u, v_fluid, u_fluid)
+    activate_particle!(system, fluid_system, particle, v, u, v_fluid, u_fluid)
 
     # Deactivate particle in interior domain
     deactivate_particle!(fluid_system, particle, u_fluid)
 
     return fluid_system
 end
 
-@inline function activate_particle!(system_new, particle_old, v_new, u_new, v_old, u_old)
+@inline function activate_particle!(system_new, system_old, particle_old,
+                                    v_new, u_new, v_old, u_old)
     particle_new = activate_next_particle(system_new)
 
-    # Exchange integrated quantities
-    for i in axes(u_new, 1)
-        u_new[i, particle_new] = u_old[i, particle_old]
-        v_new[i, particle_new] = v_old[i, particle_old]
+    # Exchange densities
```

Comment:
```suggestion
    # Transfer densities
```
This is only in one direction, so it's a transfer and not an exchange.

Shall we rename the function too? I was very confused that `activate_particle!` needs two systems. It's not just activating a particle but transferring it from one system to another. So maybe `transfer_particle!`? I'm open for suggestions.

Reply:
Very good point! `transfer_quantities!` ?

Reply:
You're actually activating and deactivating particles, so I'd prefer `transfer_particle!` to `transfer_quantities!`